packages/mobile/src/geth/saga.ts

```diff
@@ -98,6 +101,76 @@ export function* waitForGethConnectivity() {
   }
 }
 
+function* pollGethStatusUntilSynced() {
+  let head: BlockHeader = yield select(chainHeadSelector)
+
+  // Head may be null on startup
+  if (!head) {
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
+  }
+
+  while (true) {
+    if (blockIsFresh(head)) {
+      return true
+    }
+
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
+  }
+}
+
+export function* waitForGethSync() {
+  const fornoEnabled = yield select(fornoSelector)
```

Comment:
That's a good suggestion. After thinking a little more about it, I think I will leave it as is because:
1. The generator's method is more efficient than the async polling method. We can simply await an action dispatch instead of repeatedly polling. Maybe it works similarly under the hood but I think it's safe to assume it's done more efficiently than my simple polling function
2. I also like the race method better ;)